### PR TITLE
Add documentation to adb methods

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
This should be done anyway at some point of time. I will add docstring to other methods in some other PR when I have time.

### CODE REVIEW

File: lib/tools/adb-commands.js [hunk 1]
```diff
@@ -566,6 +881,11 @@ methods.restart = async function () {
   }
 };
 
+/**
+ * Start the logcat process to gather logs.
+ *
+ * @throws {error} If restart fails.
```
Comment (imurchie) [inline lib/tools/adb-commands.js]: `If start fails`

File: lib/tools/adb-commands.js [hunk 2]
```diff
@@ -686,6 +1059,16 @@ methods.instrument = async function (pkg, activity, instrumentWith) {
...
 };
 
+/**
+ * Collect Android coverage by instrumenting the particular activity.
+ *
+ * @param {string} instrumentClass - The name of the instrumentation class.
+ * @param {string} waitPkg - The name of the package to be instrumented..
+ * @param {string} waitActivity - The name of the main activity in this package.
+ *
+ * @return {promise} The promise is successfully resolved if the instrumentation starts
```
Comment (imurchie) [inline lib/tools/adb-commands.js]: This one ought to be `await`-ed as well.

File: lib/tools/adb-emu-commands.js [hunk 3]
```diff
@@ -60,17 +75,32 @@ emuMethods.fingerprint = async function (fingerprintId) {
   await this.adbExecEmu(['finger', 'touch', fingerprintId]);
 };
 
+/**
+ * Change the display orientatioo on the connected emulator.
```
Comment (imurchie) [inline lib/tools/adb-emu-commands.js]: `orientation`

File: lib/tools/adb-emu-commands.js [hunk 4]
```diff
@@ -60,17 +75,32 @@ emuMethods.fingerprint = async function (fingerprintId) {
...
+ */
 emuMethods.powerAC = async function (state = 'on') {
   if (_.values(emuMethods.POWER_AC_STATES).indexOf(state) === -1) {
     log.errorAndThrow(`Wrong power AC state sent '${state}'. Supported values: ${_.values(emuMethods.POWER_AC_STATES)}]`);
   }
   await this.adbExecEmu(['power', 'ac', state]);
 };
 
+/**
+ * Emulate power capacility change on the connected emulator.
```
Comment (imurchie) [inline lib/tools/adb-emu-commands.js]: `capacity`

File: lib/tools/android-manifest.js [hunk 5]
```diff
@@ -53,6 +53,14 @@ manifestMethods.processFromManifest = async function (localApk) {
   }
 };
 
+/**
+ * Extract package and main activity name from application manifest.
+ *
+ * @param {string} localApk - The ful path to application package.
```
Comment (imurchie) [inline lib/tools/android-manifest.js]: `full`

File: lib/tools/android-manifest.js [hunk 6]
```diff
@@ -53,6 +53,14 @@ manifestMethods.processFromManifest = async function (localApk) {
...
+ * @return {object} - the mapping containing two members: _apkPackage_ and _apkActivity_
```
Comment (imurchie) [inline lib/tools/android-manifest.js]: `The`

File: lib/tools/android-manifest.js [hunk 7]
```diff
@@ -91,6 +99,14 @@ manifestMethods.packageAndLaunchActivityFromManifest = async function (localApk)
   }
 };
 
+/**
+ * Extract target SDK version from application manifest.
+ *
+ * @param {string} localApk - The ful path to application package.
```
Comment (imurchie) [inline lib/tools/android-manifest.js]: `full`

File: lib/tools/android-manifest.js [hunk 8]
```diff
@@ -163,6 +185,12 @@ manifestMethods.insertManifest = async function (manifest, srcApk, dstApk) {
   }
 };
 
+/**
+ * Check whether package manifest contains internet permissions.
+ *
+ * @param {string} localApk - The ful path to application package.
```
Comment (imurchie) [inline lib/tools/android-manifest.js]: `full`

File: lib/tools/android-manifest.js [hunk 9]
```diff
@@ -163,6 +185,12 @@ manifestMethods.insertManifest = async function (manifest, srcApk, dstApk) {
...
+ * @return {boolean} True if the manifest requires internet access permission..
```
Comment (imurchie) [inline lib/tools/android-manifest.js]: Single period.

File: lib/tools/apk-utils.js [hunk 10]
```diff
@@ -109,6 +135,17 @@ apkUtilsMethods.getFocusedPackageAndActivity = async function () {
   }
 };
 
+/**
+ * Wait for the given activity to be focused/non-focused.
+ *
+ * @param {string} pkg - The name of the package to wait to.
```
Comment (imurchie) [inline lib/tools/apk-utils.js]: `wait for`

File: lib/tools/apk-utils.js [hunk 11]
```diff
@@ -109,6 +135,17 @@ apkUtilsMethods.getFocusedPackageAndActivity = async function () {
...
+ * @param {string} activity - The name of the activity, belonging to that package,
+ *                            to wait to.
```
Comment (imurchie) [inline lib/tools/apk-utils.js]: `wait for`

File: lib/tools/apk-utils.js [hunk 12]
```diff
@@ -213,6 +293,17 @@ apkUtilsMethods.install = async function (apk, replace = true, timeout = 60000)
   }
 };
 
+/**
+ * Install the package from the local file system of upgrade it if an older
+ * version of the same package is already installed.
+ *
+ * @param {string} apk - The full path to the local package.
+ * @param {string} pkg - The name of the installed package. The method will
+ *                       peform faster if it is set.
```
Comment (imurchie) [inline lib/tools/apk-utils.js]: `perform`

File: lib/tools/apk-utils.js [hunk 13]
```diff
@@ -302,11 +409,21 @@ apkUtilsMethods.getDeviceLanguage = async function () {
...
+ *
+ * @param {string} language - The name of the new device language.
+ */
 apkUtilsMethods.setDeviceLanguage = async function (language) {
   // this method is only used in API < 23
   await this.setDeviceSysLanguage(language);
 };
 
+/**
+ * Get the contry name of the device under test.
```
Comment (imurchie) [inline lib/tools/apk-utils.js]: `country`

File: lib/tools/apk-utils.js [hunk 14]
```diff
@@ -302,11 +409,21 @@ apkUtilsMethods.getDeviceLanguage = async function () {
...
+ *
+ * @return {string} The name of device contry.
```
Comment (imurchie) [inline lib/tools/apk-utils.js]: `country`

File: lib/tools/adb-commands.js [hunk 15]
```diff
@@ -16,20 +16,41 @@ const MAX_SHELL_BUFFER_LENGTH = 1000;
 
 let methods = {};
 
+/**
+ * Get the path to adb executable amd assign it
+ * to this.executable.path and this.executable.path properties.
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: fixed

File: lib/tools/adb-commands.js [hunk 16]
```diff
@@ -42,6 +63,12 @@ methods.getApiLevel = async function () {
   return this._apiLevel;
 };
 
+/**
+ * Retrieve the platform version of the device under test.
+ *
+ * @return (string) The platform version as a string, for example '5.0' for
+ * Android Lollipop.
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: looks like not %)

File: lib/tools/adb-commands.js [hunk 17]
```diff
@@ -51,28 +78,69 @@ methods.getPlatformVersion = async function () {
...
+ * @return {string} The output of the corresponding adb command.
+ */
 methods.clear = async function (pkg) {
   return await this.shell(['pm', 'clear', pkg]);
 };
 
+/**
+ * Grant all permissions requested by the particular package.
+ * This method is only useful on Android 6.0+ and for applications
+ * that support components-based permssionss setting.
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: fixed

File: lib/tools/adb-commands.js [hunk 18]
```diff
@@ -161,6 +241,15 @@ methods.revokePermission = async function (pkg, permission) {
   }
 };
 
+/**
+ * Retrieve the list of granted permissions for the particular package.
+ *
+ * @param {string} pkg - The package name to be processed.
+ * @param {string} cmdOutput - Op[tional parameter containing command output of
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: fixed

File: lib/tools/adb-commands.js [hunk 19]
```diff
@@ -171,6 +260,15 @@ methods.getGrantedPermissions = async function (pkg, cmdOutput = null) {
     .map((x) => x.replace(/:\sgranted=true/g, ''));
 };
 
+/**
+ * Retrieve the list of denied permissions for the particular package.
+ *
+ * @param {string} pkg - The package name to be processed.
+ * @param {string} cmdOutput - Op[tional parameter containing command output of
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: fixed

File: lib/tools/adb-commands.js [hunk 20]
```diff
@@ -171,6 +260,15 @@ methods.getGrantedPermissions = async function (pkg, cmdOutput = null) {
...
+ *                             _pm dump_ command. It speeds this mathod up
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: fixed

File: lib/tools/adb-commands.js [hunk 21]
```diff
@@ -181,6 +279,15 @@ methods.getDeniedPermissions = async function (pkg, cmdOutput = null) {
     .map((x) => x.replace(/:\sgranted=false/g, ''));
 };
 
+/**
+ * Retrieve the list of requested permissions for the particular package.
+ *
+ * @param {string} pkg - The package name to be processed.
+ * @param {string} cmdOutput - Op[tional parameter containing command output of
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: fixed

File: lib/tools/adb-commands.js [hunk 22]
```diff
@@ -181,6 +279,15 @@ methods.getDeniedPermissions = async function (pkg, cmdOutput = null) {
...
+ *                             _pm dump_ command. It speeds this mathod up
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: fixed

File: lib/tools/adb-commands.js [hunk 23]
```diff
@@ -210,12 +332,23 @@ methods.stopAndClear = async function (pkg) {
...
 methods.getTargetSdkUsingPKG = async function (pkg) {
   let stdout = await this.shell(['pm', 'dump', pkg]);
   let targetSdk = new RegExp(/targetSdk=([^\s\s]+)/g).exec(stdout)[1];
   return targetSdk;
 };
 
+/**
+ * Retrieve the list of available input methods (IMEs) for the device under test.
+ *
+ * @return {array} The list of IMEs names or an empty list.
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: fixed

File: lib/tools/adb-commands.js [hunk 24]
```diff
@@ -224,6 +357,11 @@ methods.availableIMEs = async function () {
   }
 };
 
+/**
+ * Retrieve the list of enabled input methods (IMEs) for the device under test.
+ *
+ * @return {array} The list of enabled IMEs names or an empty list.
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: fixed

File: lib/tools/adb-commands.js [hunk 25]
```diff
@@ -316,25 +493,46 @@ methods.lock = async function () {
...
 
+/**
+ * @return {string} the actual path to adb executable.
+ */
 methods.getAdbPath = function () {
   return this.executable.path;
 };
 
+/**
+ * Rertieve current screen orientation of the device under test.
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: fixed

File: lib/tools/adb-commands.js [hunk 26]
```diff
@@ -316,25 +493,46 @@ methods.lock = async function () {
...
+ *
+ * @return {number} The current orientation encoded as an integer number.
+ */
 methods.getScreenOrientation = async function () {
   let stdout = await this.shell(['dumpsys', 'input']);
   return getSurfaceOrientation(stdout);
 };
 
+/**
+ * Rertieve the screen lock state of the device under test.
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: sorry, copypaste is evil :)
all fixed

File: lib/tools/adb-commands.js [hunk 27]
```diff
@@ -367,6 +570,13 @@ methods.isSoftKeyboardPresent = async function () {
   }
 };
 
+/**
+ * Send an arbitrary Telnet command to the device under test.
+ *
+ * @param {string} command - The command to be sent.
+ *
+ * @return {promise} If resolved then contains the actual output of the given command.
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: fixed

File: lib/tools/adb-commands.js [hunk 28]
```diff
@@ -411,34 +621,54 @@ methods.sendTelnetCommand = async function (command) {
...
   return (parseInt(stdout, 10) !== 0);
 };
 
-/*
- * on: true (to turn on) or false (to turn off)
+/**
+ * Change the state of WiFi on the device under test.
+ *
+ * @param {boolean} on - True to enable and false to disable it.
+ * @param {boolean} isEmulator - Set it to true if the device under
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: fixed

File: lib/tools/adb-commands.js [hunk 29]
```diff
@@ -478,6 +722,15 @@ methods.setWifiAndData = async function ({wifi, data}, isEmulator = false) {
   }
 };
 
+/**
+ * Emulate geolocation coordinates on the device under test.
+ *
+ * @param {object} location - Location object, which should contain two properties:
+ *                            _longtitude_ and _latitude_. These should be valid
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: fixed

File: lib/tools/adb-commands.js [hunk 30]
```diff
@@ -500,19 +753,48 @@ methods.setGeoLocation = async function (location, isEmulator = false) {
   }
 };
 
+/**
+ * Remove a path recursively on the device under test.
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: added the note

File: lib/tools/adb-commands.js (hunk 2 above)
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: added await

File: lib/tools/adb-emu-commands.js (hunk 3 above)
Comment (mykola-mokhnach) [inline lib/tools/adb-emu-commands.js]: fixed

File: lib/tools/adb-emu-commands.js (hunk 4 above)
Comment (mykola-mokhnach) [inline lib/tools/adb-emu-commands.js]: fixed

File: lib/tools/android-manifest.js (hunk 5 above)
Comment (mykola-mokhnach) [inline lib/tools/android-manifest.js]: fixed

File: lib/tools/android-manifest.js (hunk 6 above)
Comment (mykola-mokhnach) [inline lib/tools/android-manifest.js]: Fixed

File: lib/tools/android-manifest.js (hunk 7 above)
Comment (mykola-mokhnach) [inline lib/tools/android-manifest.js]: fixed

File: lib/tools/android-manifest.js (hunk 8 above)
Comment (mykola-mokhnach) [inline lib/tools/android-manifest.js]: fixed

File: lib/tools/android-manifest.js (hunk 9 above)
Comment (mykola-mokhnach) [inline lib/tools/android-manifest.js]: fixed

File: lib/tools/apk-utils.js (hunk 10 above)
Comment (mykola-mokhnach) [inline lib/tools/apk-utils.js]: fixed

File: lib/tools/apk-utils.js (hunk 11 above)
Comment (mykola-mokhnach) [inline lib/tools/apk-utils.js]: fixed

File: lib/tools/apk-utils.js (hunk 12 above)
Comment (mykola-mokhnach) [inline lib/tools/apk-utils.js]: fixed

File: lib/tools/apk-utils.js (hunk 13 above)
Comment (mykola-mokhnach) [inline lib/tools/apk-utils.js]: fixed

File: lib/tools/apk-utils.js (hunk 14 above)
Comment (mykola-mokhnach) [inline lib/tools/apk-utils.js]: fixed

File: lib/tools/adb-commands.js (hunk 1 above)
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: fixed

**Pull request merged**